panther_hardware_interfaces/src/roboteq_driver.cpp

```diff
@@ -68,18 +68,16 @@ struct RoboteqCANObjects
 };
 
 RoboteqDriver::RoboteqDriver(
-  const std::shared_ptr<lely::ev::Executor> & exec,
   const std::shared_ptr<lely::canopen::AsyncMaster> & master, const std::uint8_t id,
   const std::chrono::milliseconds & sdo_operation_timeout_ms)
-: lely::canopen::FiberDriver(*exec, *master, id),
-  sdo_operation_timeout_ms_(sdo_operation_timeout_ms)
+: lely::canopen::LoopDriver(*master, id), sdo_operation_timeout_ms_(sdo_operation_timeout_ms)
 {
 }
 
 bool RoboteqDriver::Boot()
 {
   booted_.store(false);
-  return FiberDriver::Boot();
+  return LoopDriver::Boot();
 }
 
 bool RoboteqDriver::WaitForBoot()
```

Comment:
I'm just curious about what happens when we use std::future<std::string> in this boot sequence. In my opinion, it would be clearer, but I am aware of the fact that it may not be worth it. Using std::async() in CANopenController::BootDrivers() could be an option there ;)

Reply:
I'm not sure if `async` can be used in this case - `Boot` method only requests boot operation, once it completes `OnBoot` is called, for which `WaitForBoot` waits. It could be done with `future`, but I wouldn't use `string` for that - IMO that would look worse. Instead bool variable can be used, that will store result if the boot was successful or not (change `booted_` to promise). This makes sense, not sure if it will be that better, but I can change it. What do you think about it?

Reply:
Sounds good. @KmakD ?

Reply:
Sounds good to me too

Reply:
Done, 488c578836b976d4438cfc85582ea6590090557b, it wasn't even necessary to use bool promise.